aten/src/ATen/native/Distributions.cpp

```diff
@@ -219,7 +143,30 @@ Tensor _s_poisson_cpu(const Tensor& lambda, Generator *gen) {
         ret_val = sample_poisson(lambda, generator);
       }
     );
-  });
+    });
+  return ret;
+}
+
+Tensor _s_gamma_cpu(const Tensor& alpha, Generator *gen) {
+  Tensor ret = alpha.type().zeros(alpha.sizes());
+  auto alpha_ = alpha.toType(ScalarType::Double);
+  AT_DISPATCH_FLOATING_TYPES(ret.type(), "gamma", [&] {
+    THGenerator* generator = get_generator(gen);
+    CPU_tensor_apply2<scalar_t, double>(ret, alpha_,
+      [generator](scalar_t& ret_val, const double& alpha){
+        BaseSampler<double> standard_uniform([generator] () {
+	  return THRandom_standard_uniform(generator);
+	});
+        BaseSampler<double> standard_normal([generator] () {
+          return THRandom_normal(generator, 0.0, 1.0);
+        });
+        auto sample = sample_gamma<double>(alpha, standard_uniform, standard_normal);
+	ret_val = std::max(std::numeric_limits<scalar_t>::min(), (scalar_t) sample);
```

Comment:
Weird indentation again. The first `});` and `ret_val` have different levels than everything else.